### PR TITLE
[WIP] Syntax highlighting + completion support in VS

### PR DESCRIPTION
**Pull request by hmemcpy**
This PR does several things:

 1. Registers `.cake` files with the C# language service, so they could be opened directly from File -> Open
 2. Adding a custom Roslyn workspace to treat .cake files as scripts, so they aren't treated as full .cs files

This will work, trust me :)

### CODE REVIEW

File: src/cake-vs/registry.pkgdef [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿
+[$RootKey$\Languages\CodeExpansions\CSharp\Paths]
+"Snippets"="$PackageFolder$\Snippets\Cake Snippet Pack"
+
```
Comment (hmemcpy) [inline src/cake-vs/registry.pkgdef]: TODO add comments for guids

**Comment by gep13** [comment]
@hmemcpy I am loving your work!!!!

I just opened a build.cake file associated with one of my projects, and I am seeing some nice syntax highlighting for the file:

image

I am also seeing some syntax errors though, further down in the file:

image

Do we need to provide additional information to the Custom Workspace to prevent these showing up?  Or is there no way to avoid?

I can also confirm that the task snippet is also working! :smile: 

What work are you still planning for this PR?  Is there anything I can help out with?  Thanks!

**Comment by hmemcpy** [comment]
Yes, you're seeing the errors because VS thinks this is now a "valid" C# file (.cs) - because I've associated .cake files with the C# language service in registry.pkgdef.
Those errors would go away if VS would treat it as a "C# interactive script", something that happens when you open a .csx file (not scriptcs, though same extension).
Unfortunately, this distinction is based on an actual `.csx` extension, which is pretty much [hardcoded](https://github.com/dotnet/roslyn/blob/master/src/VisualStudio/Core/Def/Implementation/ProjectSystem/MiscellaneousFilesWorkspace.cs#L297-L300).

Per @jasonmalinowski's suggestion, the best way would be to essentially create my own `MiscellaneousFilesWorkspace.cs`, e.g. `CakeWorkspace.cs`, duplicating some of the logic.
Unfortunately, some things/interfaces are internal, so it would require a bit of work.

**Comment by gep13** [comment]
Closing this in favour of https://github.com/agc93/Cake.VisualStudio

**Pull request closed without merging**